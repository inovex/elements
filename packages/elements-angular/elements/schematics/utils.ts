import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
