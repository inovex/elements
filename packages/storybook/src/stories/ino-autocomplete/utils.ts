import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
