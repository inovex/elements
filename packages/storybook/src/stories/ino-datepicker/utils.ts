import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
