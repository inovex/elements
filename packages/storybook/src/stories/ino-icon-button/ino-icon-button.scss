.story-icon-button {
  .flex-parent-center {
    display: flex;
    justify-content: space-between;
    max-width: 620px;
    margin: 0 auto;
  }

  .flex-child {
    text-align: center;
  }

  h4,
  h5 {
    text-align: center;
    margin-bottom: 6px;
  }
}
