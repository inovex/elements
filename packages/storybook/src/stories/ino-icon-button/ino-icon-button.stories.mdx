import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-icon-button/readme.md";

<Meta title="Buttons/ino-icon-button/Documentation" />

<Description markdown={readme} />
