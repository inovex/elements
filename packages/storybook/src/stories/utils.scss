/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
