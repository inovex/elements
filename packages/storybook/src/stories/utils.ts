import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
