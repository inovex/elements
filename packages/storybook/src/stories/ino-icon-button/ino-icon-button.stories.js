import { boolean, select, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import findElementUpwards from '../../core/helpers/findElementUpwards';

import componentReadme from '_local-elements/src/components/ino-icon-button/readme.md';
import './ino-icon-button.scss';

import ICONS from '_local-elements/src/components/ino-icon/icons';

ICONS.push(''); // additionally: allow no icon

const eventToListen = 'click';
let managedIconButtonState = true;

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = findElementUpwards(e.target, 'ino-icon-button', 'custom');

    if (!el) {
      return;
    }

    const checkedIcons = ['star', 'favorite'];
    const uncheckedIcons = checkedIcons.map((icon) => `${icon}`);

    const elementIcon = el.getAttribute('ino-icon');

    const newIcon = checkedIcons.includes(elementIcon)
      ? `${elementIcon}`
      : uncheckedIcons.includes(elementIcon)
      ? elementIcon.substring(0, elementIcon.indexOf('_'))
      : elementIcon;

    el.setAttribute('ino-icon', newIcon);
  };

  const managedButtonHandler = function (e) {
    const el = findElementUpwards(e.target, 'ino-icon-button', 'managed');
    if (!el) {
      return;
    }

    managedIconButtonState = !managedIconButtonState;
    el.setAttribute('ino-activated', managedIconButtonState);
  };
  // == event block

  document.addEventListener(eventToListen, eventHandler);
  document.addEventListener(eventToListen, managedButtonHandler);

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener(eventToListen, eventHandler);
    document.removeEventListener(eventToListen, managedButtonHandler);
  };
}

export default {
  title: 'Buttons/<ino-icon-button>',
  parameters: {
    actions: {
      handles: [`${eventToListen} .customizable-icon-button ino-icon-button`],
    },
  },
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-icon-button">
        <style>
        .customizable-icon-button {
          --ino-icon-button-size: ${text('--ino-icon-button-size', '48px')};
          --ino-icon-button-icon-size: ${text(
            '--ino-icon-button-icon-size',
            '24px'
          )};
          --ino-icon-button-icon-color: ${text(
            '--ino-icon-button-color',
            '#000000'
          )};
          --ino-icon-button-icon-active-color: ${text(
            '--ino-icon-button-active-color',
            '#000000'
          )};
          --ino-icon-button-background-color: ${text(
            '--ino-icon-button-background-color',
            'transparent'
          )};
          --ino-icon-button-background-active-color: ${text(
            '--ino-icon-button-background-active-color',
            '#000000'
          )};
        }
      </style>
      <div class="customizable-icon-button">
        <h4>Customizable icon button</h4>
        <ino-icon-button
          class="custom"
          ino-icon="${select('ino-icon', ICONS, 'star')}"
          ino-color-scheme="${select(
            'ino-color-scheme',
            [
              '',
              'primary',
              'secondary',
              'success',
              'warning',
              'error',
              'light',
              'dark',
            ],
            ''
          )}"
          ino-activated="${boolean('ino-activated', false)}"
          ino-filled="${boolean('ino-filled', false)}"
          disabled="${boolean('disabled', false)}">
        </ino-icon-button>
      </div>

      <h4>Colors</h4>
      <div class="flex-parent-center">
        <div class="flex-child">
          <h5>Primary</h5>
          <ino-icon-button ino-icon="star" ino-color-scheme="primary"></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Secondary</h5>
          <ino-icon-button ino-icon="star" ino-color-scheme="secondary"></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Success</h5>
          <ino-icon-button ino-icon="star" ino-color-scheme="success"></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Warning</h5>
          <ino-icon-button ino-icon="star" ino-color-scheme="warning"></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Error</h5>
          <ino-icon-button ino-icon="star" ino-color-scheme="error"></ino-icon-button>
        </div>
      </div>

      <h4>Filled</h4>
      <div class="flex-parent-center">
        <div class="flex-child">
          <h5>Primary</h5>
          <ino-icon-button ino-icon="time" ino-color-scheme="primary" ino-filled></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Secondary</h5>
          <ino-icon-button ino-icon="time" ino-color-scheme="secondary" ino-filled></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Success</h5>
          <ino-icon-button ino-icon="time" ino-color-scheme="success" ino-filled></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Warning</h5>
          <ino-icon-button ino-icon="time" ino-color-scheme="warning" ino-filled></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Error</h5>
          <ino-icon-button ino-icon="time" ino-color-scheme="error" ino-filled></ino-icon-button>
        </div>
      </div>

      <h4>States</h4>
      <div class="flex-parent-center">
        <div class="flex-child">
          <h5>Activated (Managed)</h5>
          <ino-icon-button class="managed" ino-icon="time" ino-color-scheme="primary" ino-activated></ino-icon-button>
        </div>

        <div class="flex-child">
          <h5>Activated (Managed, Filled)</h5>
          <ino-icon-button class="managed" ino-icon="time" ino-color-scheme="primary" ino-filled ino-activated></ino-icon-button>
        </div>
      </div>

      <h4>Variations</h4>
      <div class="flex-parent-center">
        <div class="flex-child">
          <h5>Filled primary</h5>
          <ino-icon-button ino-filled ino-color-scheme="primary" ino-icon="time"></ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>CSS Variables</h5>
          <ino-icon-button 
            style="
              --ino-icon-button-icon-color: white;
              --ino-icon-button-background-color: purple;
              --ino-icon-button-icon-active-color: white;
              --ino-icon-button-background-active-color: pink;
            " 
            ino-icon="time"
          >
          </ino-icon-button>
        </div>
        <div class="flex-child">
          <h5>Managed + Changing colors</h5>
          <ino-icon-button
            class="managed"
            style="
              --ino-icon-button-icon-color: gray;
              --ino-icon-button-background-color: transparent;
              --ino-icon-button-icon-active-color: red;
              --ino-icon-button-background-active-color: red;
            "
            ino-icon="time"
            ino-activated
          >
          </ino-icon-button>
        </div>
      </div>
    </div>
  `;
