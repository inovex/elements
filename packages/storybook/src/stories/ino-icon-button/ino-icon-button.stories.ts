import { html } from 'lit-html';
import { defaultDecorator, withColorScheme, withIconControl } from '../utils';
import './ino-icon-button.scss';

export default {
  title: 'Buttons/ino-icon-button',
  component: 'ino-icon-button',
  decorators: [defaultDecorator],
  parameters: {
    actions: {
      handles: ['click ino-icon-button'],
    },
  },
};

export const Playground = (args) => {
  return html`
    <ino-icon-button
      activated="${args.activated}"
      disabled="${args.disabled}"
      filled="${args.filled}"
      icon="${args.icon}"
      type="${args.type}"
      autofocus="${args.autofocus}"
      color-scheme="${args.colorScheme}"
    >
    </ino-icon-button>
  `;
};
Playground.args = {
  activated: false,
  disabled: false,
  filled: false,
  autofocus: false,
};

Playground.argTypes = {
  type: {
    type: 'select',
  },
  options: ['button', 'reset', 'submit'],
};

withIconControl(Playground, 'icon', 'add');
withColorScheme(Playground, 'colorScheme', 'primary');

export const Filled = () => html`
  <div class="story-icon-button">
    <div class="flex-parent-center">
      <div class="flex-child">
        <h4>Primary</h4>
        <ino-icon-button
          icon="time"
          color-scheme="primary"
          filled
        ></ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>Secondary</h4>
        <ino-icon-button
          icon="time"
          color-scheme="secondary"
          filled
        ></ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>Success</h4>
        <ino-icon-button
          icon="time"
          color-scheme="success"
          filled
        ></ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>Warning</h4>
        <ino-icon-button
          icon="time"
          color-scheme="warning"
          filled
        ></ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>Error</h4>
        <ino-icon-button
          icon="time"
          color-scheme="error"
          filled
        ></ino-icon-button>
      </div>
    </div>
  </div>
`;

export const States = () => html`
  <div class="story-icon-button">
    <div class="flex-parent-center">
      <div class="flex-child">
        <h4>Activated (Managed)</h4>
        <ino-icon-button
          class="managed"
          icon="time"
          color-scheme="primary"
          activated
        ></ino-icon-button>
      </div>

      <div class="flex-child">
        <h4>Activated (Managed, Filled)</h4>
        <ino-icon-button
          class="managed"
          icon="time"
          color-scheme="primary"
          filled
          activated
        ></ino-icon-button>
      </div>
    </div>
  </div>
`;

export const Variations = () => html`
  <div class="story-icon-button">
    <div class="flex-parent-center">
      <div class="flex-child">
        <h4>Filled primary</h4>
        <ino-icon-button
          filled
          color-scheme="primary"
          icon="time"
        ></ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>CSS Variables</h4>
        <ino-icon-button
          style="
          --ino-icon-button-icon-color: white;
          --ino-icon-button-background-color: purple;
          --ino-icon-button-icon-active-color: white;
          --ino-icon-button-background-active-color: pink;
        "
          icon="time"
        >
        </ino-icon-button>
      </div>
      <div class="flex-child">
        <h4>Managed + Changing colors</h4>
        <ino-icon-button
          class="managed"
          style="
          --ino-icon-button-icon-color: gray;
          --ino-icon-button-background-color: transparent;
          --ino-icon-button-icon-active-color: red;
          --ino-icon-button-background-active-color: red;
        "
          icon="time"
          activated
        >
        </ino-icon-button>
      </div>
    </div>
  </div>
`;

// import readme from '../../../../../elements/src/components/ino-icon-button/readme.md';
// import { renderWithMermaid } from '../../../core/with-stencil-readme';
// export const Documentation = () => renderWithMermaid(readme);
