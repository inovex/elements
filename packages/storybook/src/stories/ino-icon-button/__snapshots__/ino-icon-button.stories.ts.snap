// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/ino-icon-button Activated smoke-test 1`] = `
<div id="root-inner">
  <div class="story-icon-button">
    <ino-icon-button activated="true"
                     disabled="false"
                     filled="false"
                     type="undefined"
                     color-scheme="primary"
                     class="ino-icon-button--color-scheme-primary hydrated"
    >
      <button class="mdc-icon-button mdc-ripple-upgraded--background-focused"
              type="undefined"
      >
        <div class="mdc-icon-button__ripple">
        </div>
        <span class="mdc-icon-button__icon">
          <ino-icon icon="add"
                    class="ino-icon--color-scheme-primary hydrated"
          >
          </ino-icon>
        </span>
      </button>
    </ino-icon-button>
  </div>
</div>
`;

exports[`Buttons/ino-icon-button Filled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-icon-button">
    <ino-icon-button activated="false"
                     disabled="false"
                     filled="true"
                     type="undefined"
                     color-scheme="primary"
                     class="ino-icon-button--color-scheme-primary ino-icon-button--filled hydrated"
    >
      <button class="mdc-icon-button ino-icon-button-filled mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
              type="undefined"
              style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
      >
        <div class="mdc-icon-button__ripple">
        </div>
        <span class="mdc-icon-button__icon">
          <ino-icon icon="add"
                    class="ino-icon--color-scheme-primary hydrated"
          >
          </ino-icon>
        </span>
      </button>
    </ino-icon-button>
  </div>
</div>
`;

exports[`Buttons/ino-icon-button Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-icon-button">
    <ino-icon-button activated="false"
                     disabled="false"
                     filled="false"
                     type="undefined"
                     color-scheme="primary"
    >
      <ino-icon icon="add">
      </ino-icon>
    </ino-icon-button>
  </div>
</div>
`;
